frontend/lib/src/components/elements/Tabs/Tabs.tsx

```diff
@@ -144,7 +148,19 @@ function Tabs(props: TabProps): ReactElement {
           }
           allTabLabels[index] = nodeLabel
 
-          const isSelected = activeTabKey.toString() === index.toString()
+          // If the tab is stale, disable it
+          const isStaleTab = isElementStale(
+            childProps.node,
+            scriptRunState,
+            scriptRunId
+          )
+          const disabled = widgetsDisabled || isStaleTab
+          // Ensure stale tab's elements are also marked stale/disabled
+          childProps.isStale = isStaleTab
+          childProps.widgetsDisabled = disabled
```

Comment:
nit: I think this can be computed above the line defining `childProps` so that we can set `childProps.isStale` and `childProps.widgetsDisabled` at definition rather than mutating the object later